java/src/jmri/jmrit/logixng/expressions/ExpressionLinuxLinePower.java

```diff
@@ -0,0 +1,236 @@
+package jmri.jmrit.logixng.expressions;
+
+import java.beans.*;
+import java.io.*;
+import java.util.*;
+
+import jmri.*;
+import jmri.jmrit.logixng.*;
+import jmri.jmrit.logixng.util.*;
+import jmri.jmrit.logixng.util.parser.*;
+import jmri.util.TimerUtil;
+
+/**
+ * Check the status of battery and power supply.
+ *
+ * @author Daniel Bergqvist Copyright (C) 2023
+ */
+public class ExpressionLinuxLinePower extends AbstractDigitalExpression
+        implements PropertyChangeListener {
+
+    private Is_IsNot_Enum _is_IsNot = Is_IsNot_Enum.Is;
+    private ProtectedTimerTask _timerTask;
+    private final int _delay = 5;
+    private boolean _lastResult;
+    private JmriException _thrownException;
+
+    public ExpressionLinuxLinePower(String sys, String user)
+            throws BadUserNameException, BadSystemNameException {
+        super(sys, user);
+
+        try {
+            _lastResult = internalEvaluate();
+        } catch (JmriException e) {
+            _thrownException = e;
+        }
+    }
+
+    @Override
+    public Base getDeepCopy(Map<String, String> systemNames, Map<String, String> userNames) throws ParserException {
+        DigitalExpressionManager manager = InstanceManager.getDefault(DigitalExpressionManager.class);
+        String sysName = systemNames.get(getSystemName());
+        String userName = userNames.get(getSystemName());
+        if (sysName == null) sysName = manager.getAutoSystemName();
+        ExpressionLinuxLinePower copy = new ExpressionLinuxLinePower(sysName, userName);
+        copy.setComment(getComment());
+        copy.set_Is_IsNot(_is_IsNot);
+        return manager.registerExpression(copy);
+    }
+
+    public void set_Is_IsNot(Is_IsNot_Enum is_IsNot) {
+        _is_IsNot = is_IsNot;
+    }
+
+    public Is_IsNot_Enum get_Is_IsNot() {
+        return _is_IsNot;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Category getCategory() {
+        return Category.LINUX;
+    }
+
+    private List<String> getLinuxPowerSupplies() throws IOException, NoPowerSuppliesException {
+        List<String> powerSupplies = new ArrayList<>();
+
+        Process process = Runtime.getRuntime().exec("upower -e");
+        try (BufferedReader buffer = new BufferedReader(new InputStreamReader(process.getInputStream())))  {
+            String line;
+            while ((line = buffer.readLine()) != null) {
+                powerSupplies.add(line);
+            }
+        }
+
+//        powerSupplies.clear();  // For testing only
```

Comment:
If this line is uncommented, the expression acts as if there is no power interfaces. That can be used to simulate the case where `upower` returns none power interfaces.